hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieLogRecordReader.java

```diff
@@ -260,7 +260,7 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
             && !HoodieTimeline.compareTimestamps(logBlock.getLogBlockHeader().get(INSTANT_TIME), HoodieTimeline.LESSER_THAN_OR_EQUALS, this.latestInstantTime
         )) {
           // hit a block with instant time greater than should be processed, stop processing further
-          break;
+          continue;
         }
```

Comment:
The reader consumption upper threshold is introduced for unnecessary reading of log block, should we drop it? I don't think so, maybe you shoud just fix the threshold itself.

Reply:
Fixing the threshold? Do you mean increase the AS_OF timestamp given from the query?

Reply:
@danny0405, after discussing with @yihua , the fix here is correct since the order of log blocks have been reserved, and the "break" logic was for the in-the-order-of-time. CC:@yihua

Reply:
The `BaseHoodieLogRecordReader` is used by the new file group reader only and this one uses `HoodieLogFormatReverseReader` which reads the log files in reverse order based on the input log file list.  Right now for time travel queries, the file system view still returns more logs than required (e.g., bf, log_t1, log_t2, log_t3, log_t4, for time travel query with `as of t2`), so this change makes sure the processing does not stop unnecessarily before the file system view returns the exact files to process.

@linliu-code could you add this detail to the other ticket fixing the file system view APIs based on completion time, so we should revisit this part alongside the FSV fix?